src/NzbDrone.Core/Http/CloudFlare/CloudFlareHttpInterceptor.cs

```diff
@@ -23,12 +26,19 @@ public HttpRequest PreRequest(HttpRequest request)
 
         public HttpResponse PostResponse(HttpResponse response)
         {
+            //ToDo: Determine if the ChallengeScript is still valid and update if needed
             if (response.StatusCode == HttpStatusCode.Forbidden && response.Content.Contains(_cloudFlareChallengeScript))
             {
                 _logger.Debug("CloudFlare CAPTCHA block on {0}", response.Request.Url);
                 throw new CloudFlareCaptchaException(response, CreateCaptchaRequest(response));
             }
 
+            if (response.StatusCode == HttpStatusCode.ServiceUnavailable && IsCloudflareProtected(response))
```

Comment:
One question, and this may require testing, but is this going to catch and throw before we get to Flaresolverr processing?